core/templates/components/profile-link-directives/profile-link-image-backend-api.service.ts

```diff
@@ -0,0 +1,47 @@
+// Copyright 2018 The Oppia Authors. All Rights Reserved.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//      http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS-IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+/**
+ * @fileoverview Service to get profile image url.
+ */
+
+import { downgradeInjectable } from '@angular/upgrade/static';
+import { HttpClient } from '@angular/common/http';
+import { Injectable } from '@angular/core';
+
+interface IProfileResponse {
+  'profile_picture_data_url_for_username': string;
+}
+
+@Injectable({
+  providedIn: 'root'
+})
+export class ProfileLinkImageBackendApiService {
+  constructor(
+    private http: HttpClient
+  ) {}
+
+  fetchProfilePicture(profileImageUrl: string): Promise<string> {
+    return this.http.get(profileImageUrl).toPromise().then(
```

Comment:
The .then part should be in the component.

Reply:
Done!